src/map/script.c

```diff
@@ -21172,6 +21172,21 @@ BUILDIN_FUNC(adopt)
 	return SCRIPT_CMD_FAILURE;
 }
 
+/*=======================================================
+* Returns weapon level of item calling this command.
+* Returns 0 when caller isn't weapon or error.
+* This command accepts no argument. [secretdataz]
+*-------------------------------------------------------*/
+BUILDIN_FUNC(getweaponlv)
+{
+	TBL_PC *sd;
+	if ((sd = script_rid2sd(st)) != NULL && sd->inventory_data[current_equip_item_index])
```

Comment:
must be value check before using `current_equip_item_index`,
`current_equip_item_index > -1 && current_equip_item_index < EQI_MAX`